File: libs/eave-stdlib-py/src/eave/stdlib/eave_origins.py
import enum


class EaveApp(enum.StrEnum):
    eave_api = "eave_api"
    eave_dashboard = "eave_dashboard"
    eave_metabase = "eave_metabase"


class ExternalOrigin(enum.StrEnum):
    github_api_client = "github_api_client"

File: libs/eave-stdlib-ts/src/eave-origins.ts
import { InvalidEnumCase } from "./exceptions.js";

export enum EaveApp {
  eave_api = "eave_api",
  eave_dashboard = "eave_dashboard",
  eave_github_app = "eave_github_app",
}

export function appengineServiceName(eaveApp: EaveApp): string {
  switch (eaveApp) {
    case EaveApp.eave_api:
      return "api";
    case EaveApp.eave_dashboard:
      return "dashboard";
    case EaveApp.eave_github_app:
      return "github";
    default:
      throw new InvalidEnumCase(eaveApp);
  }
}

export enum ExternalOrigin {
  github_api_client = "github_api_client",
}
